From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ayu Kashyap <you@example.com>
Date: Wed, 22 Oct 2025 00:00:00 +0530
Subject: [PATCH] Haptics: Use brightness file and increase duration

Change haptics logic to use /sys/class/leds/vibrator/brightness
as this is the correct path for this device.
Increase haptics duration to 200ms as 50ms is too short.

Change-Id: I8b06ac64a1c7f2bb79a1295c505950afbf46a3d3
---
 minuitwrp/events.cpp | 21 ++++++++++++++++++---
 1 file changed, 18 insertions(+), 3 deletions(-)

diff --git a/minuitwrp/events.cpp b/minuitwrp/events.cpp
index b5e5a14..050a453 100644
--- a/minuitwrp/events.cpp
+++ b/minuitwrp/events.cpp
@@ -63,8 +63,8 @@ static std::atomic_int vib_on_count = 0;
 #define VIBRATOR_TIMEOUT_FILE  "/sys/class/timed_output/vibrator/enable"
 #define VIBRATOR_TIME_MS    50
 
-#define LEDS_HAPTICS_DURATION_FILE "/sys/class/leds/vibrator/duration"
-#define LEDS_HAPTICS_ACTIVATE_FILE "/sys/class/leds/vibrator/activate"
+/* Use brightness file for haptics */
+#define LEDS_HAPTICS_BRIGHTNESS_FILE "/sys/class/leds/vibrator/brightness"
 
 #ifndef SYN_REPORT
 #define SYN_REPORT          0x00
@@ -78,16 +78,31 @@ static void vibrate(int)
             ret = 0;
         }
     } else {
-        int fd = open(LEDS_HAPTICS_DURATION_FILE, O_RDWR);
+        // Use brightness file logic
+        int fd = open(LEDS_HAPTICS_BRIGHTNESS_FILE, O_RDWR);
         if (fd >= 0) {
-            write(fd, "50", 2);
+            write(fd, "255", 3); // Vibrate ON (full brightness)
             close(fd);
-            fd = open(LEDS_HAPTICS_ACTIVATE_FILE, O_RDWR);
+            
+            // Wait for 200ms (200 * 1000 microseconds)
+            // VIBRATOR_TIME_MS (50ms) is too short to be felt.
+            usleep(200 * 1000);
+            
+            fd = open(LEDS_HAPTICS_BRIGHTNESS_FILE, O_RDWR);
             if (fd >= 0) {
-                write(fd, "1", 1);
+                write(fd, "0", 1); // Vibrate OFF
                 close(fd);
-                ret = 0;
             }
+            ret = 0; // Signal success
         }
     }
 

-- 
2.39.2